scripts/pre_push_hook.py

```diff
@@ -122,10 +146,10 @@ def _collect_files_being_pushed(ref_list, remote):
         ref_list: list of references to parse (provided by git in stdin)
         remote: the remote being pushed to
     Returns:
```

Comment:
`git push REMOTE - -all` pushes everything (and I mean everything) this is the only reason that this got more complicated than it needs to be :wink:

regarding your second point,  I do not quite get what you mean? We do only lint the branches that are pushed